homeassistant/components/kraken/sensor.py

```diff
@@ -141,7 +141,7 @@ def _update_internal_state(self):
                     )
                     self._available = False
 
-    def _try_get_state(self) -> str:
+    def _try_get_state(self) -> str:  # type: ignore[return]
```

Comment:
If we add a class attribute for the coordinator and type that we don't need to ignore here.

Eg: https://github.com/home-assistant/core/blob/3200b0150af6c275de229cf22ce3bf9ab9e36be5/homeassistant/components/gios/air_quality.py#L60

Reply:
It complains here because it does not return anything when a `TypeError` is thrown. This can be the case when an asset got delisted or no data at all was returned.

I could merge `_try_get_state` into `_update_internal_state` to avoid this. But I like them separated to reduce the cognitive load.

Reply:
You can raise an exception in the end if nothing matched.